### PR TITLE
EOS-26572: Updates related to s3 services name changes in provisioner test framework

### PR DESCRIPTION
**Pull request by tanujashinde0405**
# Problem Statement
- EOS-26572: Updates related to s3 services name changes in provisioner test framework

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by nileshgovande** [comment]
Thanks @tanujashinde0405  for branch in provisioner repo

Hello @nitin-seagate 

Can you please help me update below page with these updated names (as I do not have access rights)

ConfigMap Confluence:
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/611615649/CORTX+Configuration+Parameters

We are also adding the names of the services in POD definition files for the startup script as well

s3 currently supports both old names as well as new names for old deployment (2 PODs) as well as redefined deployment where motr and s3 are separate - We have validated this by doing a deployment.

**Review by nileshgovande — APPROVED** [review]

File: test/deploy/kubernetes/runtime-pods/components_control_node.yaml [hunk 1]
```diff
@@ -102,14 +102,14 @@ spec:
           mountPath: /share
         - name: local-node1
           mountPath: /etc/cortx
-      - name: cortx-bg-producer
+      - name: cortx-bg-scheduler
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/runtime-pods/components_control_node.yaml:105]: The format for container names should be: `<component>-<service>`
For example: `s3-bgscheduler`

File: test/deploy/kubernetes/runtime-pods/components_server_node1.yaml [hunk 2]
```diff
@@ -168,14 +168,14 @@ spec:
           name: "port-9443"
         - containerPort: 28049
           name: "port-28049"
-      - name: cortx-s3bgc
+      - name: cortx-s3-bgworker
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/runtime-pods/components_server_node1.yaml:171]: The format for container names should be: `<component>-<service>`
For example: `s3-bgworker`

File: test/deploy/kubernetes/template-config/cluster_default.yaml [hunk 3]
```diff
@@ -11,9 +11,10 @@ cluster:
       - name: hare
       - name: s3
         services:
-        - io
-        - auth
-        - bg_consumer
+        - s3server
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/template-config/cluster_default.yaml:14]: Why are changing in old infra? Will it work for 2 pod setup (Control and Storage PODs)

File: test/deploy/kubernetes/runtime-pods/components_server_node2.yaml [hunk 4]
```diff
@@ -185,4 +185,4 @@ spec:
         - name: shared-node2
           mountPath: /share
         - name: local-node2
-          mountPath: /etc/cortx
+          mountPath: /etc/cortx
```
Comment (nitin-seagate) [inline test/deploy/kubernetes/runtime-pods/components_server_node2.yaml:188]: Add new line

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: test/deploy/kubernetes/template-config/cluster_default.yaml (hunk 3 above)
Comment (tanujashinde0405) [reply test/deploy/kubernetes/template-config/cluster_default.yaml:14]: yes , it is working for 2 pod set up also, we have tested it.

File: test/deploy/kubernetes/runtime-pods/components_control_node.yaml (hunk 1 above)
Comment (tanujashinde0405) [reply test/deploy/kubernetes/runtime-pods/components_control_node.yaml:105]: in Chirag's Pr we tried to follow services framework format , and they have cortx-<component>-<services> 
should we revert back to <component>-<service> then?

Comment (nitin-seagate) [reply test/deploy/kubernetes/runtime-pods/components_control_node.yaml:105]: Lets take it as part of redefined pods infra...

File: test/deploy/kubernetes/runtime-pods/components_server_node1.yaml (hunk 2 above)
Comment (nitin-seagate) [reply test/deploy/kubernetes/runtime-pods/components_server_node1.yaml:171]: Lets take it as part of redefined pods infra...

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Pull request merged**